modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/ContentDashboardItemFactoryTracker.java

```diff
@@ -67,20 +90,174 @@ protected void activate(BundleContext bundleContext) {
 				bundleContext, ContentDashboardItemFactory.class, null,
 				ServiceReferenceMapperFactory.create(
 					bundleContext,
-					(contentDashboardItem, emitter) -> emitter.emit(
+					(contentDashboardItemFactory, emitter) -> emitter.emit(
 						GenericUtil.getGenericClassName(
-							contentDashboardItem))));
+							contentDashboardItemFactory))),
+				new ContentDashboardItemFactoryTrackerCustomizer(
+					bundleContext));
 	}
 
 	@Deactivate
 	protected void deactivate() {
 		_serviceTrackerMap.close();
 	}
 
+	private void _addAssetVocabulary(
+			Company company, long classNameId, String name, int visibilityType)
+		throws Exception {
+
+		User defaultUser = company.getDefaultUser();
+
+		Map<Locale, String> titleMap = new HashMap<>();
+
+		for (Locale locale :
+				_language.getCompanyAvailableLocales(company.getCompanyId())) {
+
+			titleMap.put(locale, LanguageUtil.get(locale, name));
+		}
+
+		AssetVocabularySettingsHelper assetVocabularySettingsHelper =
+			new AssetVocabularySettingsHelper();
+
+		assetVocabularySettingsHelper.setClassNameIdsAndClassTypePKs(
+			new long[] {classNameId},
+			new long[] {AssetCategoryConstants.ALL_CLASS_TYPE_PK},
+			new boolean[] {false});
+
+		ServiceContext serviceContext = new ServiceContext();
+
+		serviceContext.setAddGroupPermissions(true);
+		serviceContext.setAddGuestPermissions(true);
+
+		_assetVocabularyLocalService.addVocabulary(
+			null, defaultUser.getUserId(), company.getGroupId(), name,
+			StringPool.BLANK, titleMap, Collections.emptyMap(),
+			assetVocabularySettingsHelper.toString(), visibilityType,
+			serviceContext);
+	}
+
+	private void _addClassNameIdToVocabularyIfNotExist(
+		long classNameId, String name) {
+
+		try {
+			_companyLocalService.forEachCompany(
+				company -> {
+					AssetVocabulary assetVocabulary =
+						_assetVocabularyLocalService.fetchGroupVocabulary(
+							company.getGroupId(),
+							StringUtil.toLowerCase(GetterUtil.getString(name)));
+
+					if (assetVocabulary == null) {
+						_addAssetVocabulary(
+							company, classNameId, name,
+							AssetVocabularyConstants.VISIBILITY_TYPE_INTERNAL);
+
+						return;
+					}
+
+					AssetVocabularySettingsHelper
+						assetVocabularySettingsHelper =
+							new AssetVocabularySettingsHelper();
+
+					long[] selectedClassNameIds =
```

Comment:
Are we retrieving here the entities associated to this vocabulary, like BlogsEntry, JournalArticles, and so on?

Reply:
The classNameIds of these classes, yes.